plugins/woocommerce/src/Utilities/NumberUtil.php

```diff
@@ -49,4 +53,85 @@ public static function array_sum( array $arr ): float {
 
 		return array_sum( $sanitized_array );
 	}
+
+	/**
+	 * Sanitize a cost value based on the current locale decimal and thousand separators.
+	 *
+	 * @param string $value               The value to sanitize.
+	 * @return string                     The sanitized value.
+	 * @throws \InvalidArgumentException If the value is not a valid numeric value.
+	 */
+	public static function sanitize_cost_in_current_locale( $value ): string {
+		$value                      = is_null( $value ) ? '' : $value;
+		$value                      = wp_kses_post( trim( wp_unslash( $value ) ) );
+		$currency_symbol_encoded    = get_woocommerce_currency_symbol();
+		$currency_symbol_variations = array( $currency_symbol_encoded, wp_kses_normalize_entities( $currency_symbol_encoded ), html_entity_decode( $currency_symbol_encoded ) );
+
+		$value = str_replace( $currency_symbol_variations, '', $value );
+
+		// Count the number of decimal points.
+		$decimal_point_count = substr_count( $value, '.' );
+
+		// If it's a standard decimal number (single decimal point and is_numeric), accept it directly. This could be in the case where the frontend has de-localised the value.
+		// We check for the decimal point count because is_numeric is looser than that.
+		if ( 1 === $decimal_point_count && is_numeric( $value ) ) {
+			return $value;
+		}
+
+		// Otherwise, attempt to delocalise according to localisation rules.
+		$allowed_characters_regex = sprintf(
+			'/^[0-9\%s\%s]*$/',
+			wc_get_price_thousand_separator(),
+			wc_get_price_decimal_separator()
+		);
+
+		if ( 1 !== preg_match( $allowed_characters_regex, $value ) ) {
+			throw new \InvalidArgumentException(
+				esc_html(
+					sprintf(
+						/* translators: %1$s: Invalid value that was input by the user, %2$s: thousand separator, %3$s: decimal separator */
+						__( '%1$s is not a valid numeric value. Allowed characters are numbers, and the thousand (%2$s) and decimal (%3$s) separators.', 'woocommerce' ),
```

Comment:
I believe Matt generally pushes for Oxford commas which (AFAIK) just means we should add a `,` before `and` in this case because it's "a list" 😄 

```suggestion
						__( '%1$s is not a valid numeric value. Allowed characters are numbers, the thousand (%2$s), and decimal (%3$s) separators.', 'woocommerce' ),
```